cargo-dist/src/tasks.rs

```diff
@@ -49,50 +92,221 @@ pub const CPU_ARM64: &str = "arm64";
 /// The key for referring to 32-bit arm as an "cpu"
 pub const CPU_ARM: &str = "arm";
 
+/// A rust target-triple (e.g. "x86_64-pc-windows-msvc")
+pub type TargetTriple = String;
+/// A map where the order doesn't matter
+pub type FastMap<K, V> = std::collections::HashMap<K, V>;
+/// A map where the order matters
+pub type SortedMap<K, V> = std::collections::BTreeMap<K, V>;
+/// A set where the order matters
+pub type SortedSet<T> = std::collections::BTreeSet<T>;
+
 /// Contents of METADATA_DIST in Cargo.toml files
-#[derive(Deserialize, Debug)]
-pub struct DistMetadata {}
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct DistMetadata {
+    /// The intended version of cargo-dist to build with. (normal Cargo SemVer syntax)
+    ///
+    /// When generating full tasks graphs (such as CI scripts) we will pick this version.
+    ///
+    /// FIXME: Should we produce a warning if running locally with a different version? In theory
```

Comment:
I think warning would be a good idea, for two reasons:

1. When someone runs `cargo-dist` locally, then most likely they want to test what the CI will do. But if the CI uses a different version then the user locally, they may see different behavior which could cause confusion. A heads up that the local build may not produce the same result because versions differ would solve that.
2. This version here essentially functions as a lock file, ensuring CI builds are reproducible. The problem is that users have to remember to bump this version when a new `cargo-dist` release is published. We could therefore use that opportunity to inform the user that their CI is still using an older version and that they can update.

Reply:
I largely agree, it's mostly a question of working out the right UX for it